### PR TITLE
Offer contributed data as bulk file for download 

### PR DESCRIPTION
**Pull request by metincansiper**
Refs #717

### CODE REVIEW

File: src/util/document.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import fetch from 'node-fetch';
```
Comment (jvwong) [inline src/util/document.js:1]: I'm not sure this file is necessary. Also node-fetch may be possible issue if used in browser?

File: src/util/document.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+import { tryPromise } from './promise';
+
+function convertDocumentToBiopax(docId, baseUrl = '') {
+  let SERVER_URL = baseUrl + `/api/document/biopax/${docId}`;
+  let makeRequest = () => fetch(SERVER_URL);
```
Comment (jvwong) [inline src/util/document.js:7]: `node-fetch` won't differentiate a response based on HTTP status. So you could get garbage in the response. I made a `util/fetch.js` helper for this.

File: src/util/document.js (hunk 1 above)
Comment (jvwong) [reply src/util/document.js:1]: > I'm not sure this file is necessary.

Ok I see you have a client-side module.

File: src/util/document.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  return tryPromise( makeRequest ).then( result => result.text() );
+}
+
+function convertDocumentToTxt(docId, baseUrl = '') {
```
[the remark this answers is not in the document]
Comment (metincansiper) [reply src/util/document.js:11]: Okay I can exclude txt from the bulk filw. However, I guess you do not mean to remove ``exportDocumentToTxt()`` in ``client-util`` as well? n that case I think that this function would stay just to be called from client side and would not be included in the bulk file.

**Comment by metincansiper** [comment]
@jvwong I made the requested changes. For now I excluded the text conversion just from the bulk file.

File: src/util/document.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (jvwong) [reply src/util/document.js:12]: > Okay I can exclude txt from the bulk filw. 

Fine with me.

File: src/util/document.js (hunk 1 above)
Comment (maxkfranz) [reply src/util/document.js:1]: I'm pretty sure `fetch()` should just be available globally on the server.  Doing separate `import`s each time fetch is used on the server could make for more refactoring work later if we ever have to swap out the polyfill.

**Comment by maxkfranz** [comment]
Looks good to me.  

Did we decide on a final name for the route?  Is there a reason to use `/api/document/zip/dl` versus `/api/document/zip`?

**Comment by metincansiper** [comment]
>Did we decide on a final name for the route? Is there a reason to use /api/document/zip/dl versus /api/document/zip?

When I tried to make it ``/api/document/zip`` it did not work because I guess that it conflicted with another service like ``/api/document/${id}`. where id is ``zip``.

**Comment by jvwong** [comment]
Re the route, you can just move it up. 
Also can you add the Swagger for this route?

File: src/util/document.js (hunk 1 above)
Comment (metincansiper) [reply src/util/document.js:1]: It gives an error by telling that fetch is not defined in server side if I do not import fetch. I tried that it still works in client side as well when I import the node-fetch.

**Comment by metincansiper** [comment]
>Re the route, you can just move it up.

I could not understand this. Do you mean adjusting it as ``api/zip`` or ``api/export/zip``? If so actually ``document`` was very convenient place in terms of implementation. 

>Also can you add the Swagger for this route?

Sure I  can do that.

**Comment by metincansiper** [comment]
>I could not understand this. Do you mean adjusting it as ``api/zip`` or ``api/export/zip``? If so actually ``document`` was very convenient place in terms of implementation. 

Okay I guess you mean just moving the function up in the file before the conflicting service. Looks like that works I can do that.

**Comment by metincansiper** [comment]
>Okay I guess you mean just moving the function up in the file before the conflicting service. Looks like that works I can do that.

@jvwong but would it be 100% safe? I cannot imagine any doument with an id like "zip" but would not it still be a confusion?

**Comment by metincansiper** [comment]
I added the swagger for the route I can rename it along with relocating the service in the file after the final decision is made on that.

**Comment by maxkfranz** [comment]
The ids are uuids. The only exception is the special ‘demo’ case

**Comment by jvwong** [comment]
@metincansiper  I'm going to push an update to the Swagger doc description...

**Comment by metincansiper** [comment]
>@metincansiper I'm going to push an update to the Swagger doc description...

Okay, sounds good.

>The ids are uuids. The only exception is the special ‘demo’ case

Okay then I will relocate and rename the service as Jeff suggested.

**Comment by metincansiper** [comment]
@maxkfranz @jvwong  Okay, I relocated the api to ``api/document/zip``.

**Comment by jvwong** [comment]
Works for me.

**Pull request merged**